Testnet/Ticketbooth/Ticketbooth/TicketContract_1_0_0.cs

```diff
@@ -0,0 +1,474 @@
+    {
+        Assert(Message.Sender == Owner, "Only contract owner can set identity verification policy");
+        Assert(!SaleOpen, "Sale is open");
+        RequireIdentityVerification = requireIdentityVerification;
+    }
+
+    /// <summary>
+    /// Requests a refund for a ticket, which will be issued if the no refund block limit is not yet reached
+    /// </summary>
+    /// <param name="seatIdentifierBytes">The serialized seat identifier</param>
+    public void ReleaseTicket(byte[] seatIdentifierBytes)
+    {
+        Assert(SaleOpen, "Sale not open");
+        Assert(Block.Number + NoRefundBlockCount < EndOfSale, "Surpassed no refund block limit");
+
+        var seat = Serializer.ToStruct<Seat>(seatIdentifierBytes);
+        var copyOfTickets = Tickets;
+        Ticket ticket = default(Ticket);
+        int ticketIndex = 0;
+        for (var i = 0; i < copyOfTickets.Length; i++)
+        {
+            if (SeatsAreEqual(copyOfTickets[i].Seat, seat))
+            {
+                ticketIndex = i;
+                ticket = copyOfTickets[i];
+                break;
+            }
+        }
+
+        Assert(!IsDefaultSeat(ticket.Seat), "Seat not found");
+        Assert(Message.Sender == ticket.Address, "You do not own this ticket");
+
+        if (ticket.Price > ReleaseFee)
+        {
+            Transfer(Message.Sender, ticket.Price - ReleaseFee);
+        }
+
+        copyOfTickets[ticketIndex].Address = Address.Zero;
+        copyOfTickets[ticketIndex].Secret = null;
+        copyOfTickets[ticketIndex].CustomerIdentifier = null;
+        Tickets = copyOfTickets;
+
+        Log(copyOfTickets[ticketIndex]);
+    }
+
+    private Ticket SelectTicket(byte[] seatIdentifierBytes)
+    {
+        var seat = Serializer.ToStruct<Seat>(seatIdentifierBytes);
+        foreach (var ticket in Tickets)
+        {
+            if (SeatsAreEqual(ticket.Seat, seat))
+            {
+                return ticket;
+            }
+        }
+
+        return default(Ticket);
+    }
+
+    private bool IsAvailable(Ticket ticket)
+    {
+        return ticket.Address == Address.Zero;
+    }
+
+    private bool IsDefaultSeat(Seat seat)
+    {
+        return seat.Number == default(int) || seat.Letter == default(char);
+    }
+
+    private bool SeatsAreEqual(Seat seat1, Seat seat2)
+    {
+        return seat1.Number == seat2.Number && seat1.Letter == seat2.Letter;
+    }
+
+    private Ticket[] ResetTickets(Ticket[] tickets)
+    {
+        for (int i = 0; i < tickets.Length; i++)
+        {
+            tickets[i].Price = 0;
+            tickets[i].Address = Address.Zero;
+            tickets[i].Secret = null;
+            tickets[i].CustomerIdentifier = null;
+        }
+
+        return tickets;
+    }
+
+    /// <summary>
+    /// Identifies a specific seat by number and/or letter
+    /// </summary>
+    public struct Seat
+    {
+        /// <summary>
+        /// A number identifying the seat
+        /// </summary>
+        public int Number;
```

Comment:
Just to be safe from negative values, would you want to use uint here instead?

Reply:
I don't see any reason to prevent this. You never know, a venue might use negative numbers for some ridiculous reason!

Reply:
If that's the case, would you want to be extra safe and use a string instead? In case the seat number is something like `A23`?

Reply:
The seat struct already allows a letter. But I see your point. Now I think about it, a byte is a better choice to reduce storage cost.

Reply:
Everything is stored as bytes under the hood. Sounds like you already thought this through. All good here.